src/test/java/winstone/HttpConnectorFactoryTest.java

```diff
@@ -49,6 +56,29 @@ public void testListenAddress() throws Exception {
 
     @Test
     public void writePortInFile() throws Exception {
+        Callable<String> asyncVerifier = () -> {
+            System.out.println("Async Verifier started!");
+
+            while (true) {
```

Comment:
if there is a bug and the file is not written we will rely on surefire timeout which can be a bit long.
at least we need some timeout but yup http://www.awaitility.org/ will be better.
see comment here https://github.com/jenkinsci/winstone/pull/233#issuecomment-1165099862